### DISCUSSION TITLE
Buttons do NOT work.. solution :)

### DISCUSSION THREAD

**Discussion by gilmod714** [acemielektron/fddEMU#32] [Ideas]
First off, Thank you so much Acemi for this wonderful solution .. it's a good project and Great work on the code too! -- it is basically a poor man's GOTEK :)

This project worked GREAT for me because I already had a failed Gotek device/enclosure. (it was a bad STM32 chip and replacing it was costly) .. So your design was perfect to renovate that old enclosure .. Only issue I had was, the buttons did not work at first  (the first 2 buttons worked , but the others did not) .. serial debug showed activity ok though 

SOLUTION:
So after some digging and looking at your code .. I figured it was some sort of a timing issue in the ladder of buttons .. I ended up changing the ADCVAL in "ADCButton.cpp" and recompiled the code (see attached screen shot) .. (also note: to recompile nano or nano-copy properly you need to add -b 57600 to avrdude .. add to make-file or do manually)

It worked great after that .. I also removed "Eject", just removed the button and kept the resistor.. inserting a new IMG replaces the old one anyways and the Gotek case had room for only 4 buttons crunched in the front on a perf PCB ;) .. 

Just posting this to help someone who have the same issue with the buttons ..

Photos .. 

![IMG_20240330_193711](https://github.com/acemielektron/fddEMU/assets/163355639/41c69e4a-c142-47a5-8e7b-375bb1f3a37c)
![IMG_20240330_193734](https://github.com/acemielektron/fddEMU/assets/163355639/972438be-10ef-4bd8-94e9-1b2a135a8a35)
![IMG_20240330_193938](https://github.com/acemielektron/fddEMU/assets/163355639/049a8d8e-3515-458d-98fb-f0a026e4687d)

The simplified schematic (for my build) ..

![fddEMU_schematic_v1](https://github.com/acemielektron/fddEMU/assets/163355639/9c76a8f4-e867-48fe-b27b-9ef2d8a887f8)

**Comment by acemielektron** [comment]
Thanks, I plan to mention this in readme

**Reply by gilmod714** [reply]
Keep up the good work!
